runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5279,6 +5279,9 @@ class INTERPRETER_CLASS
 		yieldContinuation(_currentThread);
 
 		j9object_t continuationObject = J9VMJAVALANGTHREAD_CONT(_currentThread, _currentThread->carrierThreadObject);
+		if(J9VMJDKINTERNALVMCONTINUATION_FINISHED(_currentThread, continuationObject)) {
+			freeContinuation(_currentThread, continuationObject);
```

Comment:
Is this going to cause any issues with the GC call below (i.e. does the GC call need the native struct for any reason)?

Reply:
Also missing space after if.

Reply:
Their are some related changes GC's concurrent changeset #16713 . From what I am aware, this doesn't impact GC call as they will verify the vmRef pointer before scanning.
@LinHu2016 Can you provide more insight?

Reply:
For last unmount continuation, GC should not call postUnmountContinuation for the compensating scan(there is an update in https://github.com/eclipse-openj9/openj9/pull/16713), anyway there is a protection checking vmRef pointer before scanning, so it should not cause any issues for the GC scan after  freeContinuation(), but there is a race case between unmounting and concurrent scanning, which we also fixed in https://github.com/eclipse-openj9/openj9/pull/16713, above the change would neither help the case nor cause more issues.

Reply:
OK, please still fix the space.

Reply:
`freeContinuation` frees the native memory and NULLs the object field. For concurrent GC, what prevents the invalid use of the native memory pointer (GC thread reads the field as another thread is about to free it)? @LinHu2016

Reply:
Concurrent GC will try to atomically win the right to scan by setting a scan bit in continuation state, if (among other conditions) continuation is not finished.
Finished is one of the flags in continuation state and it's set during last unmount before the struct is freed and pointer reset.
On the other side last unmount (setting finished flag in the continuation state) cannot occur while concurrent scan is in progress (in fact even a preceding mount cannot occur, and will wait till scan is complete).

Reply:
There are no atomics in `freeContinuation` so that code is presumably somewhere in the callers.

Reply:
Indeed, freeContinuaiton (or anywhere around unmount path) there are no atomics associated with this synchronization. As mentioned, in the context of carrier thread, it is earlier during mounting that must have occurred before: in synchronizeWithConcurrentGCScan.

In the context of concurrent GC thread, synchronization/atomics occur in tryWinningConcurrentGCScan

However, I'm coming to realization that continuation state should really be a part (a hidden field) of Continuation Java object (btw, I even suggested in past that whole Continuation struct could really be an extension of Java object, mostly for other reasons, reducing stress on native heap etc).
As we have it now, seems like concurrent GC thread doing atomics in tryWinningConcurrentGCScan while racing with last unmount, could operate on the state field that could just been freed (and in very unlikely scenario be reallocated by unrelated party that could initialize that same piece of memory so that it satisfies all the conditions necessary to concurrent scan to win, while it should not).

Reply:
This really has nothing to do with this PR, I just thought about it when I saw the code. We should probably open a new issue to track the timing hole. I think putting the state in the java object is the only way to address the free timing hole, as you say.

Reply:
I will open new issue about the free timing hole between checking if NULL != continuation and checking continuation->state.

Reply:
freeContinuation change tracked in https://github.com/fengxue-IS/openj9/tree/freeContinuation